File: java/src/main/java/com/example/cote/PGMRS/가장_큰_수.java
package com.example.cote.PGMRS;

import java.util.*;
import java.util.stream.*;

public class 가장_큰_수 {

  private static final int TEST_CASES = 10;
  private static final Random rd = new Random(43);
  private static final int N = 1000000;
  private static final int M = 1000;

  private static int[] initialize() {

    int n = rd.nextInt(N - 1) + 1;
    int[] ret = new int[n];

    for (int i = 0; i < n; i++) ret[i] = rd.nextInt(M);

    return ret;
  }

  private static String solve(int[] numbers) {

    return Arrays.stream(numbers)
        .mapToObj(Integer::toString)
        .sorted((a, b) -> -(a + b).compareTo(b + a))
        .collect(Collectors.joining(""))
        .replaceAll("^0+", "0");
  }

  public static void main(String... args) {

    for (int it = 0; it < TEST_CASES; it++) {
      System.out.println(solve(initialize()));
    }
  }
}

File: c++/cote/PGMRS/가장큰수.cpp
#include <iostream>
#include <random>
#include <string>

#define TEST_CASES 10
#define N 100

using namespace std;

vector<int> initialize(default_random_engine& gen) {
  vector<int> ret;
  uniform_int_distribution<int> dis(0, 100);
  int n = abs(dis(gen)) % 5 + 5;

  for (int i = 0; i < n; i++) ret.push_back(abs(dis(gen)));
  return ret;
}

string solve(vector<int>& n) {
  vector<string> s;

  for (int i = 0; i < n.size(); i++) s.push_back(to_string(n[i]));

  sort(begin(s), end(s), [](string a, string b) { return a + b > b + a; });

  string ret;
  for (int i = 0; i < s.size(); i++) ret += s[i];
  return ret;
}

int main(void) {
  default_random_engine gen(43);

  for (int it = 0; it < TEST_CASES; it++) {
    vector<int> n = initialize(gen);

    for (int i = 0; i < n.size(); i++) cout << n[i] << " ";
    cout << endl;

    cout << solve(n) << endl;
  }
  return 0;
}